examples/files.rs

```diff
@@ -0,0 +1,218 @@
+use std::collections::HashSet;
+use std::ffi::OsString;
+use std::path::{Path, PathBuf};
+use std::time::{Duration, Instant};
+
+use crossterm::event::{Event, KeyCode, MouseEventKind};
+use ratatui::backend::{Backend, CrosstermBackend};
+use ratatui::layout::Rect;
+use ratatui::style::{Color, Modifier, Style};
+use ratatui::text::{Line, Span};
+use ratatui::widgets::{Block, Scrollbar, ScrollbarOrientation};
+use ratatui::{Frame, Terminal};
+use tui_tree_widget::{Node, Tree, TreeData, TreeState};
+
+struct FileTreeData(PathBuf);
+
+impl TreeData for FileTreeData {
+    type Identifier = OsString;
+
+    fn flatten(&self, open: &HashSet<Vec<Self::Identifier>>) -> Vec<Node<Self::Identifier>> {
```

Comment:
Thinking through the ergonomics of some of this (and reading it as someone without actual tree widget experience, perhaps `open_identifiers` would be a better name than `open`, and I wonder if there's a good way of making a specific type or type alias for `Vec<Self::Identifier>` that makes it clear that acts like a path to the identifier - something like `type IdentifierPaths<T> = HashSet<Vec<T>>`, or an actual real type for this? (I'm talking paths to the tree nodes not file paths).

Reply:
I thought about having a dedicated identifier trait instead of `Vec<Identifier>`.

`PathBuf` would allow for everything that is needed without needing the `Vec<OsString>` workaround.
Same with JSON paths that have their own RFC.

Reply:
Interesting idea. I guess that could work.